torch/hub.py

```diff
@@ -209,6 +212,68 @@ def _get_cache_or_reload(github, force_reload, verbose=True, skip_validation=Fal
 
     return repo_dir
 
+def _get_trusted_input(repo):
+    # separate function for testing
+    response = input(
+        f"The repository {repo} does not belong to the list of trusted repositories and as such cannot be downloaded. "
+        "Do you trust this repository and wish to add it to the trusted list of repositories (y/N)?")
+    return response
+
+def _check_repo(repo_owner, repo_name, repo_branch, trust_repo, calling_fn="load"):
+    hub_dir = get_dir()
+    filepath = os.path.join(hub_dir, "trusted_list")
+    filepath_legacy = os.path.join(hub_dir, "trusted_list_legacy")
+
+    if not os.path.exists(filepath):
+        Path(filepath).touch()
+
+    if not os.path.exists(filepath_legacy):
+        # Initialize legacy file with all existing repos
+        repos = next(os.walk(hub_dir))[1]
+        with open(filepath_legacy, "w") as file:
+            for _repo in repos:
+                if _repo != "checkpoints":
+                    file.write(_repo + "\n")
+
+    # load list
+    with open(filepath, 'r') as file:
+        trusted_repos = tuple(line.strip() for line in file if len(line.strip()))
+    with open(filepath_legacy, 'r') as file:
+        trusted_repos_legacy = tuple(line.strip() for line in file if len(line.strip()))
+    repo = '_'.join([repo_owner, repo_name])
+    repo_branch = '_'.join([repo_owner, repo_name, repo_branch])
```

Comment:
Nit: it might be easier to follow what's going with:

```suggestion
    owner_name = '_'.join([repo_owner, repo_name])
    owner_name_branch = '_'.join([repo_owner, repo_name, repo_branch])
```